engine/graphics/proto/graphics/graphics_ddf.proto

```diff
@@ -84,6 +84,14 @@ enum DepthStencilFormat
     DEPTH_STENCIL_FORMAT_S8U      = 5;
 }
 
+// Engine only, values corresponds to TextureUsageHint in graphics.h
+enum TextureUsageFlag
+{
+    TEXTURE_USAGE_FLAG_SAMPLE     = 1;
+    TEXTURE_USAGE_FLAG_MEMORYLESS = 2;
+    TEXTURE_USAGE_FLAG_STORAGE    = 4;
+}
```

Comment:
These are only here because we create dynamic textures via the resource system, which requires DDF data for creating. We could potentially avoid this by adding an extra function to set it after the HTexture has been created, but it requires unique functions per adapters, and a bit strange program flow in the case where a texture has been created already and someone tries to change the hint, which isn't possible in any API.